File: Py Scripts/GameControl.py
import UnityEngine
import UnityEngine.UI

class GameControl(UnityEngine.MonoBehaviour):
    def Start(self):
        self.starter = self.GetComponent(CountDown)
        self.scoreTextLeft = UnityEngine.UI.Text()
        self.scoreTextRight = UnityEngine.UI.Text()
        self.scoreLeft = 0
        self.scoreRight = 0
        self.ball = UnityEngine.GameObject()
        self.ballControl = self.ball.GetComponent(BallScript)
        self.started = False
        self.startingPosition = UnityEngine.Vector3()

    def Update(self):
        if self.started:
            return
        
        if UnityEngine.Input.GetKey(UnityEngine.KeyCode.Space):
            self.started = True
            self.starter.StartCountdown()

    def StartGame(self):
        self.ballControl.Go()

    def ScoreGoalLeft(self):
        self.scoreRight += 1
        self.UpdateUI()
        self.ResetBall()

    def ScoreGoalRight(self):
        self.scoreLeft += 1
        self.UpdateUI()
        self.ResetBall()

    def UpdateUI(self):
        self.scoreTextLeft.text = str(self.scoreLeft)
        self.scoreTextRight.text = str(self.scoreRight)

    def ResetBall(self):
        self.ballControl.Stop()
        self.ball.transform.position = self.startingPosition
        self.starter.StartCountdown()
        self.ball.SetActive(True)

File: scripts/GameControl.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class GameControl : MonoBehaviour
{
    public CountDown starter;
        
    public Text scoreTextLeft;
    public Text scoreTextRight;
    private int scoreLeft = 0;
    private int scoreRight = 0;
    public GameObject ball;
    public BallScript ballControl;
    public bool started = false;
    public Vector3 startingPosition;
    // Start is called before the first frame update
    void Start()
    {
        
        
        this.startingPosition = this.ball.transform.position;

    }

    // Update is called once per frame
    void Update()
    {
        if ( this.started )
         return;

      
        if (Input.GetKey(KeyCode.Space))
        {
            this.started = true;
            this.starter.StartCountdown();
        }

    }


    public void StartGame()
    {
        this.ballControl.Go();
    }
    public void ScoreGoalLeft()
    {
        this.scoreRight += 1;
        UpdateUI();
        ResetBall();
    }
    public void ScoreGoalRight()
    {
        this.scoreLeft += 1;
        UpdateUI();
        ResetBall();
    }
    private void UpdateUI()
    {
        this.scoreTextLeft.text = this.scoreLeft.ToString();
        this.scoreTextRight.text = this.scoreRight.ToString();
    }

    private void ResetBall()
    {
        this.ballControl.Stop();
        this.ball.transform.position = this.startingPosition;
        this.starter.StartCountdown();
       
        
        this.ball.SetActive(true);
    }
   

}
